sync/class.jetpack-sync-listener.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+require_once dirname( __FILE__ ) . '/class.jetpack-sync-queue.php';
+require_once dirname( __FILE__ ) . '/class.jetpack-sync-functions.php';
+require_once dirname( __FILE__ ) . '/class.jetpack-sync-defaults.php';
+
+require_once dirname( __FILE__ ) . '/class.jetpack-sync-modules.php';
+
+/** 
+ * This class monitors actions and logs them to the queue to be sent
+ */
+class Jetpack_Sync_Listener {
+
+	private $sync_queue;
+
+	// singleton functions
+	private static $instance;
+
+	public static function getInstance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	// this is necessary because you can't use "new" when you declare instance properties >:(
+	protected function __construct() {
+		$this->set_defaults();
+		$this->init();
+	}
+
+	private function init() {
+
+		$handler = array( $this, 'action_handler' );
+
+		foreach( Jetpack_Sync_Modules::get_modules() as $module ) {
+			$module->init_listeners( $handler );
+		}
+
+		// synthetic actions for full sync
+		add_action( 'jetpack_full_sync_start', $handler );
+		add_action( 'jetpack_full_sync_end', $handler );
+		
+		// Module Activation
```

Comment:
Probably - I didn't move it yet because I still wasn't sure what to do with the full sync code in general... I tried not to touch it too much in this refactor. Do you think we should try to make that change here, or in a separate PR to clean up full sync?